### PR TITLE
Fixes user experience bits with SQL docs and code sample

### LINKED ISSUE #216: Retouch spring-cloud-gcp-sql-example README
- Add pointer to the Cloud SQL Cloud Console page.
- Add clarity to when Cloud SDK is enabled and project ID/creds can be auto-discoverable vs application.properties need to be set.
- Need to talk about "spring.datasource.username" and "spring.datasource.password" in the refdoc
- Clarify everywhere that Cloud SQL API must be enabled (and not any of the other Cloud SQL copycats in the APIs screen).

### PR DESCRIPTION
**Pull request by joaoandremartins**
Fixes #216

### CODE REVIEW

**Comment by joaoandremartins** [comment]
This contains an important fix to the SQL reference document (explicitly saying where the SQL username and password should go) that it would be nice to get in before the M1 release cut.

File: spring-cloud-gcp-docs/src/main/asciidoc/sql.adoc [hunk 1]
```diff
@@ -29,6 +29,18 @@ dependencies {
...
+
+In order to use the Spring Boot Starter for Google Cloud SQL features, the Google Cloud SQL API
+must be enabled in your GCP project.
+
+To do that, go to the
+https://console.cloud.google.com/apis/library[API library page] of the Google Cloud Console, search
+for "Cloud SQL API", click the first result and enabled the API.
+
+NOTE: There are several similar "Cloud SQL" results. You must access the "Google Cloud SQL API" one
+and enable the API from there
```
Comment (meltsufin) [inline spring-cloud-gcp-docs/src/main/asciidoc/sql.adoc]: Period at the end of the sentence.

File: spring-cloud-gcp-docs/src/main/asciidoc/sql.adoc [hunk 2]
```diff
@@ -29,6 +29,18 @@ dependencies {
...
 
 
+=== Prerequisites
+
+In order to use the Spring Boot Starter for Google Cloud SQL features, the Google Cloud SQL API
+must be enabled in your GCP project.
+
+To do that, go to the
+https://console.cloud.google.com/apis/library[API library page] of the Google Cloud Console, search
+for "Cloud SQL API", click the first result and enabled the API.
```
Comment (meltsufin) [inline spring-cloud-gcp-docs/src/main/asciidoc/sql.adoc]: "and enabled" -> ", and enable"

File: spring-cloud-gcp-starters/spring-cloud-gcp-starter-sql/README.adoc [hunk 3]
```diff
@@ -25,6 +25,12 @@ This starter provides the configuration sources `CloudSqlJdbcInfoProvider` and `
 It only allows connections to second-generation Google Cloud SQL instances.
 
 Before using the starter, ensure that the Google Cloud SQL API is enabled in your GCP project.
+To do that, go to the
+https://console.cloud.google.com/apis/library[API library page] of the Google Cloud Console, search
+for "Cloud SQL API", click the first result and enabled the API.
+
+NOTE: There are several similar "Cloud SQL" results. You must access the "Google Cloud SQL API" one
+and enable the API from there
```
Comment (meltsufin) [inline spring-cloud-gcp-starters/spring-cloud-gcp-starter-sql/README.adoc]: Period at the end.

File: spring-cloud-gcp-starters/spring-cloud-gcp-starter-sql/README.adoc [hunk 4]
```diff
@@ -25,6 +25,12 @@ This starter provides the configuration sources `CloudSqlJdbcInfoProvider` and `
 It only allows connections to second-generation Google Cloud SQL instances.
 
 Before using the starter, ensure that the Google Cloud SQL API is enabled in your GCP project.
+To do that, go to the
+https://console.cloud.google.com/apis/library[API library page] of the Google Cloud Console, search
+for "Cloud SQL API", click the first result and enabled the API.
```
Comment (meltsufin) [inline spring-cloud-gcp-starters/spring-cloud-gcp-starter-sql/README.adoc]: "and , enable"

**Pull request merged**